src/app/core/parameters/segmentArray.ts

```diff
@@ -0,0 +1,29 @@
+import Parameter from "../parameter";
+import Registry from "../registry";
+import pointArray from "./pointArray"
+
+export default class SegmentArray extends Parameter {
+    static readonly typeString: string = "SegmentArray";
+    static readonly description: string = "Should be an array of PointArrays";
+
+    constructor(value: Array<Array<Array<number>>>) {
```

Comment:
Similarly, we should add standard types here

Reply:
Should I create a .d.ts file for the parameters/ type definitions?

Reply:
Yup, we can change it once @jackruan99 finishes it up in his repos